tests/monitoring/monitoring.go

```diff
@@ -320,6 +327,67 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", func() {
 		})
 	})
 
+	Context("VM status metrics", func() {
+		newVirtualMachine := func() *v1.VirtualMachine {
+			vmi := tests.NewRandomVMI()
+			return tests.NewRandomVirtualMachine(vmi, false)
+		}
+
+		createVirtualMachine := func(vm *v1.VirtualMachine) {
+			By("Creating VirtualMachine")
+			_, err := virtClient.VirtualMachine(util.NamespaceTestDefault).Create(vm)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		BeforeEach(func() {
+			scales = make(map[string]*autoscalingv1.Scale, 1)
+			backupScale(virtOperator.deploymentDame)
+			updateScale(virtOperator.deploymentDame, 0)
+
+			reduceAlertPendingTime()
+		})
+
+		AfterEach(func() {
+			revertScale(virtOperator.deploymentDame)
+
+			waitUntilAlertDoesNotExist("KubeVirtVMStuckInStartingState")
+			waitUntilAlertDoesNotExist("KubeVirtVMStuckInErrorState")
+		})
+
+		It("KubeVirtVMStuckInStartingState should be triggered if VM is taking more than 5 minutes to start", func() {
+			vm := newVirtualMachine()
+			vm.Spec.Template.Spec.PriorityClassName = "non-preemtible"
+			createVirtualMachine(vm)
+
+			By("Starting VirtualMachine")
+			err = virtClient.VirtualMachine(util.NamespaceTestDefault).Start(vm.Name, &v1.StartOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			verifyAlertExist("KubeVirtVMStuckInStartingState")
+
+			err = virtClient.VirtualMachine(util.NamespaceTestDefault).Delete(vm.GetName(), &metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Please avoid deleting VMs in the end of the test. They would be removed automatically.
If objects are being removed manually it prevents us from collecting relevant test information at the end of the test.

Reply:
removed